internal/tracinginterceptor/interceptor.go

```diff
@@ -39,54 +45,189 @@ var (
+		ParentSpanContext: parentSpanCtx,
+		Tracer:            i.tracer,
+		TransportName:     req.Transport,
+		StartTime:         time.Now(),
+		ExtraTags:         commonTracingTags,
+	}
+	ctx, span := extractOpenTracingSpan.Do(ctx, req)
+	defer span.Finish()
+
+	err := h.HandleOneway(ctx, req)
+	return updateSpanWithErrorDetails(span, false, nil, err)
 }
 
 // CallOneway implements interceptor.OnewayOutbound
-func (m *Interceptor) CallOneway(ctx context.Context, request *transport.Request, out transport.OnewayOutbound) (transport.Ack, error) {
-	// TODO: implement
-	panic("implement me")
+func (i *Interceptor) CallOneway(ctx context.Context, req *transport.Request, out transport.OnewayOutbound) (transport.Ack, error) {
+	createOpenTracingSpan := &transport.CreateOpenTracingSpan{
+		Tracer:        i.tracer,
+		TransportName: i.transport,
+		StartTime:     time.Now(),
+		ExtraTags:     commonTracingTags,
+	}
+	ctx, span := createOpenTracingSpan.Do(ctx, req)
+	defer span.Finish()
+
+	tracingHeaders := make(map[string]string)
+	if err := i.tracer.Inject(span.Context(), i.propagationFormat, getPropagationCarrier(tracingHeaders, i.transport)); err != nil {
+		span.LogFields(logFieldEventError, log.String("message", err.Error()))
+	} else {
+		for k, v := range tracingHeaders {
+			req.Headers = req.Headers.With(k, v)
+		}
+	}
+
+	ack, err := out.CallOneway(ctx, req)
+	return ack, updateSpanWithErrorDetails(span, false, nil, err)
 }
 
 // HandleStream implements interceptor.StreamInbound
-func (m *Interceptor) HandleStream(s *transport.ServerStream, h transport.StreamHandler) error {
-	// TODO: implement
+func (i *Interceptor) HandleStream(s *transport.ServerStream, h transport.StreamHandler) error {
 	panic("implement me")
 }
 
 // CallStream implements interceptor.StreamOutbound
-func (m *Interceptor) CallStream(ctx context.Context, req *transport.StreamRequest, out transport.StreamOutbound) (*transport.ClientStream, error) {
-	// TODO: implement
+func (i *Interceptor) CallStream(ctx context.Context, req *transport.StreamRequest, out transport.StreamOutbound) (*transport.ClientStream, error) {
 	panic("implement me")
 }
+
+func updateSpanWithErrorDetails(
+	span opentracing.Span,
+	isApplicationError bool,
+	appErrorMeta *transport.ApplicationErrorMeta,
+	err error,
+) error {
+	if err == nil && !isApplicationError {
+		return nil
+	}
+	ext.Error.Set(span, true)
+	if status := yarpcerrors.FromError(err); status != nil {
+		errCode := status.Code()
+		span.SetTag(rpcStatusCodeTag, int(errCode))
+		span.SetTag(errorCodeTag, errCode.String())
+		return err
+	}
+	if isApplicationError {
+		span.SetTag(errorCodeTag, applicationErrorTag)
+
+		if appErrorMeta != nil {
+			if appErrorMeta.Code != nil {
+				span.SetTag(rpcStatusCodeTag, int(*appErrorMeta.Code))
+			}
+			if appErrorMeta.Name != "" {
+				span.SetTag(errorNameTag, appErrorMeta.Name)
+			}
+		}
+		return err
+	}
+	span.SetTag(errorCodeTag, unknownInternalYarpcError)
```

Comment:
We have two sort of error code span tags here: 1) `rpc.yarpc.status_code`, 2) `error.code`. 
And we have 3 conditions here: 1) L198 - yarpc code derivable from err, 2) L204 - application err, 3) L217 - none of above

But they are not consistent:
* condition 1. sets tag 1) and tag 2)
* condition 2. sets tag 1)
* condition 3. sets tag 2)

Let's just just use tag 1) - `rpc.yarpc.status_code` for all of them.  So for condition 3, can we use 
```
span.SetTag(rpcStatusCodeTag, int(yarpcerrors.CodeUnknown))
```
And for condition 1, let's remove `error.code` tag.


For this general `error.code` span tag, let's add it during our ingestion pipeline, and use similar format like muttley error code: https://engwiki.uberinternal.com/display/TE0OBSERVAPPS/RPC+Error+Codes